### PR TITLE
[Feature] Build Physician Assessment Edit Modal

### PR DESCRIPTION
**Pull request by j-bunk**
## Notion ticket link

[Build Physician Assessment edit modal](https://www.notion.so/uwblueprintexecs/Build-Physician-Assessment-edit-modal-337e42efe73f4bd59ec1a98b2415ea38)

## Implementation description
Added Physician Assessment Edit Modal and mobility aids section to card + modal

- [x] Build edit modal
- [x] Add `UPDATE_PHYSICIAN_ASSESSMENT` mutation
- [x] Hook up modal to card
  - [x] Hook up modal + add mutation call
  - [x] Add mobility section to card (missed in previous PR)
  - [x] Add `permitType` to `NewApplication` table (confirm with @OustanDing)
- [x] Add mobility aids section to form component (unrelated but must have been missed when the form component was built)
  - [x] Build section
  - [x] Update db to include `CRUTCHES` as a mobility aid, and also add `otherMobilityAids` field
- [x] Address NOTETOSELF comment(s)

## Demo

https://user-images.githubusercontent.com/51224641/167760884-11e99933-c7b3-4bdc-9db3-aca293aaa7b7.mov

## Notes
* I noticed that the mobility aids section was missing from the Physician Assessment card and form components and added it to them in this PR
* Form validation needs to be added and the edit modal will need to be refactored accordingly (needs to be wrapped in a Formik component). I think this can be addressed in the form validation PR once this is merged in.
* Will merge after https://github.com/uwblueprint/richmond-centre-for-disability/pull/202 is merged

## Checklist
- [x] My PR name is descriptive, is in imperative tense and starts with one of the following: `[Feature]`,`[Improvement]` or `[Fix]`,
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the RCD team on GitHub, or specific people who are associated with this ticket

### CODE REVIEW

File: components/admin/requests/additional-questions/EditModal.tsx [hunk 1]
```diff
@@ -62,7 +62,7 @@ export default function EditAdditionalInformationModal({
               paddingX="4px"
             >
               <Text as="h2" textStyle="display-medium-bold">
-                {'Edit Payment, Shipping and Billing Details'}
+                {'Edit Additional Information'}
```
Comment (j-bunk) [inline components/admin/requests/additional-questions/EditModal.tsx:65]: Unrelated to this PR

File: components/admin/requests/physician-assessment/Card.tsx [hunk 2]
```diff
@@ -108,6 +168,23 @@ const Card: FC<Props> = props => {
...
+            <List>
+              {mobilityAids?.map(aid => (
+                <ListItem key={aid}>
+                  <Text as="p" textStyle="body-regular" textAlign="left">
+                    {titlecase(aid)}
+                  </Text>
+                </ListItem>
+              ))}
+            </List>
+          </GridItem>
```
Comment (j-bunk) [inline components/admin/requests/physician-assessment/Card.tsx:187]: I must have missed this when building the card. It's in some of the specs and missing in others, I'll confirm with designers if this section is conditionally rendered

File: components/admin/requests/physician-assessment/Form.tsx [hunk 3]
```diff
@@ -121,6 +133,79 @@ export default function PhysicianAssessmentForm({
...
+              }}
+            >
+              <Stack>
+                <Radio value={'1'}>{'Yes, I am'}</Radio>
+                <Radio value={'0'}>{'No, I am not'}</Radio>
+              </Stack>
+            </RadioGroup>
+          </FormControl>
+
+          {mobilityAidsRequired === true && (
```
Comment (j-bunk) [inline components/admin/requests/physician-assessment/Form.tsx:165]: Lmk if there's a cleaner way to handle this. The current approach is to use a state `mobilityAidsRequired` to determine whether or not to render the Other mobility aids text area. If the user selects that they do not have mobility aids the if statement is hit and the value of mobility aid is set to `[]`.

File: lib/applications/schema.ts [hunk 4]
```diff
@@ -750,6 +752,8 @@ export default gql`
     mobilityAids: [MobilityAid!]
     otherPatientCondition: String
     temporaryPermitExpiry: Date
+    permitType: PermitType!
```
Comment (j-bunk) [inline lib/applications/schema.ts:755]: I think `permitType` needs to be included to allow users to change the `permitType` using the `EditModal` but it seems to have been purposely omitted here (there's a comment on line 748 saying to omit it).

image

File: prisma/schema.prisma [hunk 5]
```diff
@@ -246,8 +246,10 @@ model NewApplication {
   disabilityCertificationDate          DateTime                             @map("disability_certification_date") @db.Date
   patientCondition                     PatientCondition                     @map("patient_condition")
   mobilityAids                         MobilityAid[]                        @map("mobility_aids")
+  otherMobilityAids                    String?                              @map("other_mobility_aids") @db.VarChar(255)
```
Comment (j-bunk) [inline prisma/schema.prisma:249]: Added `otherMobilityAids` to match the design

image

File: prisma/schema.prisma [hunk 6]
```diff
@@ -246,8 +246,10 @@ model NewApplication {
...
   otherPatientCondition                String?                              @map("other_patient_condition") @db.VarChar(255)
   temporaryPermitExpiry                DateTime?                            @map("temporary_permit_expiry") @db.Date
+  permitType                           PermitType                           @default(PERMANENT) @map("permit_type")
```
Comment (j-bunk) [inline prisma/schema.prisma:252]: See above https://github.com/uwblueprint/richmond-centre-for-disability/pull/192/files#r869824458

File: prisma/schema.prisma [hunk 7]
```diff
@@ -440,6 +442,8 @@ enum MobilityAid {
   MANUAL_CHAIR
   SCOOTER
   WALKER
+  CRUTCHES
+  OTHERS
```
Comment (j-bunk) [inline prisma/schema.prisma:446]: Added `CRUTCHES` and `OTHERS` to match the design

image

File: tools/admin/requests/additional-questions.ts [hunk 8]
```diff
@@ -56,7 +56,7 @@ export type GetAdditionalInformationResponse = {
 
 /** Update additional information of application */
 export const UPDATE_ADDITIONAL_INFORMATION = gql`
-  mutation UpdateAdditionalInformationInformation(
+  mutation UpdateApplicationAdditionalInformation(
```
Comment (j-bunk) [inline tools/admin/requests/additional-questions.ts:59]: Unrelated to this PR

**Review by carelynntsai — APPROVED** [review]
😍

File: components/admin/requests/physician-assessment/Card.tsx (hunk 2 above)
Comment (j-bunk) [reply components/admin/requests/physician-assessment/Card.tsx]: Resolved!

File: components/admin/requests/physician-assessment/Card.tsx [hunk 9]
```diff
@@ -108,6 +170,41 @@ const Card: FC<Props> = props => {
...
+                  Other mobility aids
+                </Text>
+              </GridItem>
+              <GridItem>
+                <Text as="p" textStyle="body-regular" textAlign="left">
+                  {otherMobilityAids}
+                </Text>
+              </GridItem>
+            </>
+          )}
```
Comment (j-bunk) [inline components/admin/requests/physician-assessment/Card.tsx:207]: Confirming with designers if they want this (see slack thread https://uwblueprint.slack.com/archives/C021S4WGHBR/p1652067821964179)

**Review by OustanDing — CHANGES_REQUESTED** [review]

File: components/admin/requests/physician-assessment/Form.tsx (hunk 3 above)
Comment (j-bunk) [reply components/admin/requests/physician-assessment/Form.tsx]: Addressed!

File: components/admin/requests/physician-assessment/EditModal.tsx [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+};
+
+export default function EditPhysicianAssessmentModal({
+  children,
+  physicianAssessment: currentPhysicianAssessment,
+  onSave,
+}: Props) {
+  const { isOpen, onOpen, onClose } = useDisclosure();
+  const [physicianAssessment, setPhysicianAssessment] = useState<PhysicianAssessment>(
+    currentPhysicianAssessment || {
```
[the remark this answers is not in the document]
Comment (j-bunk) [reply components/admin/requests/physician-assessment/EditModal.tsx]: Addressed!

File: prisma/schema.sql [hunk 11]
```diff
@@ -383,8 +385,10 @@ CREATE TABLE new_applications (
   disability_certification_date DATE NOT NULL,
   patient_condition PatientCondition NOT NULL,
   mobility_aids MobilityAid ARRAY,
+  other_mobility_aids VARCHAR(255),
   other_patient_condition VARCHAR(255),
   temporary_permit_expiry DATE,
+  permit_type PermitType NOT NULL DEFAULT 'PERMANENT',
```
[the remark this answers is not in the document]
Comment (j-bunk) [reply prisma/schema.sql]: Addressed in latest commit!

File: lib/applications/schema.ts (hunk 4 above)
Comment (j-bunk) [reply lib/applications/schema.ts:755]: Would you mind clarifying here? Is there anything I should look out for 👀

Comment (j-bunk) [reply lib/applications/schema.ts:777]: @OustanDing bump^

**Comment by j-bunk** [comment]
@OustanDing Now that https://github.com/uwblueprint/richmond-centre-for-disability/pull/202 is merged in, this PR can be merged into the larger form validation PR https://github.com/uwblueprint/richmond-centre-for-disability/pull/158.

I've addressed all your PR feedback except for https://github.com/uwblueprint/richmond-centre-for-disability/pull/192#discussion_r870883630, once that's resolved I'll merge this PR in

File: lib/applications/schema.ts (hunk 4 above)
Comment (OustanDing) [reply lib/applications/schema.ts:777]: Yes, to clarify, your change here is correct, so no further action is needed in this file. What I meant by backwards compatibility is that we should ensure that any other API calls that depend on this input schema should still work.

Comment (j-bunk) [reply lib/applications/schema.ts:777]: Sweet, tested the api the relevant api calls and everything looks good!

**Comment by j-bunk** [comment]
@OustanDing address the comment! PR should be good to merge if things look good to you!

**Review by OustanDing — APPROVED** [review]

**Pull request merged**